Feature/comment : 댓글 작성, 수정, 삭제, 조회

reeruryu opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
**AS-IS**
- 댓글 작성
=> 알람 추가
- 수정, 삭제
- 조회
=> 조회부분만 유저 체크를 하지 않았습니다.
- 유저 체크를 우선 코드처럼 했는데 어떻게 해야 할지.. 코멘트 주시면 감사하겠습니다!

**TO-BE**
- 우선 제가 하기로 한 건 다 했는데, 도와드릴 거나 해야 할 것들 말씀해 주시면 감사하겠습니다.

### 테스트 
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [x] 테스트 코드
- [x] API 테스트

reeruryu committed:
feat : 댓글 - 작성, 수정, 삭제, 목록조회

reeruryu committed:
test : CommentService 테스트 추가

reeruryu committed:
Merge branch 'develop' into feature/comment

reeruryu committed:
fix : 빠진 콤마(,) 추가

reeruryu committed:
feat : 댓글 작성, 삭제 시 리뷰의 댓글 수 change

sunju5402 commented on src/main/java/com/example/seatchoice/service/CommentService.java:
```diff
@@ -0,0 +1,97 @@
+
+		String commentsUrl =
+			"localhost:8080/api/review/" + review.getId() + "/comments";
+
```
배포할 때는 저희 주소로 해야하지 않을까요??

sunju5402 commented on src/main/java/com/example/seatchoice/dto/param/CommentParam.java:
```diff
@@ -0,0 +1,41 @@
+		@NotNull(message = "필수 입력입니다.")
+		private Long reviewId;
+
+		@NotBlank(message = "댓글을 작성해 주세요.")
```
최대 어느정도까지 작성할 수 있는지도 검증하면 좋을 거 같습니다!

sunju5402 commented on src/main/java/com/example/seatchoice/dto/param/CommentParam.java:
```diff
@@ -0,0 +1,41 @@
+	@Getter
+	@AllArgsConstructor
+	public static class Create {
+		@NotNull(message = "유저 id를 입력해주세요.")
```
userId는 controller쪽에서 @AuthenticationPrincipal User user 쓰셔서 user 정보를 받아오는 것도 
좋을 거 같습니다!
이 부분은 정아님이랑 얘기해보면 좋을 거 같아요

reeruryu replied:
네넵 코멘트 감사합니다 ㅎㅎ👍!!

sunju5402 reviewed (commented):
희주님 수고하셨어요!

reeruryu committed:
chore : pr 코멘트 반영

- commentsUrl에서 localhost -> ec2 ip 주소 수정
- 댓글 작성, 수정 시 최소 길이 5자로 검증 추가 @NotBlank -> @Pattern
- memberId를 OAuth2User로 받아오도록 수정
- 바뀐 부분로 인한 테스트 코드 수정
